scirpy/io/_datastructures.py

```diff
@@ -234,17 +236,28 @@ def _key_sort_chains(chain) -> Tuple:
         return tuple(-1 if x is None else x for x in sort_tuple)
 
     @staticmethod
-    def _serialize_chains(chains: List[MutableMapping]) -> str:
+    def _serialize_chains(
+        chains: List[MutableMapping], include_fields: Optional[Collection[str]] = None
+    ) -> str:
         """Serialize chains into a JSON object. This is useful for storing
         an arbitrary number of extra chains in a single column of a dataframe."""
         # convert numpy dtypes to python types
         # https://stackoverflow.com/questions/9452775/converting-numpy-dtypes-to-native-python-types
         for chain in chains:
             for k, v in chain.items():
-                try:
-                    chain[k] = chain[k].item()
-                except AttributeError:
-                    pass
+                if include_fields is not None:
+                    if k in include_fields:
+                        try:
+                            chain[k] = chain[k].item()
+                        except AttributeError:
+                            pass
+                    else:
+                        pass
```

Comment:
```suggestion
```

Reply:
this looks the same? is this just a black formatting thing? if so, i'll just make sure to format before my next push.

Reply:
Actually, I was suggesting to remove it entirely as it does nothing ;)

Reply:
ah of course. misread the colours.